pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/services/PluginCheckerService.kt

```diff
@@ -5,105 +5,185 @@ package com.amos.pitmutationmate.pitmutationmate.services
 
 import com.amos.pitmutationmate.pitmutationmate.plugincheck.PluginCheckerGroovy
 import com.amos.pitmutationmate.pitmutationmate.plugincheck.PluginCheckerKotlin
+import com.amos.pitmutationmate.pitmutationmate.ui.ToolWindowFactory
 import com.intellij.openapi.components.Service
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.diagnostic.thisLogger
+import com.intellij.openapi.fileEditor.FileDocumentManager
 import com.intellij.openapi.project.Project
-import com.intellij.openapi.ui.Messages
-import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.ToolWindowManager
 import com.intellij.psi.PsiManager
+import com.intellij.util.ui.UIUtil
 import org.apache.commons.io.IOUtils
 import org.codehaus.groovy.ast.builder.AstBuilder
-import java.io.File
-import java.io.FileInputStream
+import java.io.ByteArrayInputStream
+import java.io.InputStream
 
 @Service(Service.Level.PROJECT)
 class PluginCheckerService(private val project: Project) {
+    private val logger = Logger.getInstance(this::class.java)
 
```

Comment:
```suggestion
    fun reset() {
        isKotlinPitestPluginAvailable = false
        isKotlinCompanionPluginAvailable = false
        isGroovyPitestPluginAvailable = false
        isGroovyCompanionPluginAvailable = false
    }
```

Reply:
they get overwritten every time a new check is ran

Reply:
I have refactored the way the booleans are set. Have a look.